lib/msal-node/src/client/ManagedIdentitySources/AzureArc.ts

```diff
@@ -84,36 +125,53 @@ export class AzureArc extends BaseManagedIdentitySource {
         const [identityEndpoint, imdsEndpoint] =
             AzureArc.getEnvironmentVariables();
 
-        // if either of the identity or imds endpoints are undefined, this MSI provider is unavailable.
+        // if either of the identity or imds endpoints are undefined
```

Comment:
Can this check be avoided? At this point it is already determined that the source is azure arc managed identity. You should now just create the instance of azure arc managed identity. This method should validate the endpoint URI and create an instance of azure arc managed identity. I think we should have done similar for all the sources.

Reply:
No, we need it for logging. We do this logging for all MI sources.